pkg/osv/osv.go

```diff
@@ -306,26 +306,35 @@ func HydrateWithClient(resp *BatchedResponse, client *http.Client) (*HydratedBat
 	return &hydrated, nil
 }
 
-// makeRetryRequest will return an error on both network errors, and if the response is not 200
+// makeRetryRequest will return an error on both network errors, and if the response is not 2xx
 func makeRetryRequest(action func() (*http.Response, error)) (*http.Response, error) {
 	var resp *http.Response
 	var err error
 
-	for i := range maxRetryAttempts {
```

Comment:
issue: I think you've changed this loop more than you need to

 - `for i := range maxRetryAttempts`  should be equivalent to what you've got currently
 - you've removed the random jitter, which is intentionally present to [smooth the back-off](https://stackoverflow.com/questions/46939285/why-is-random-jitter-applied-to-back-off-strategies)

Reply:
Yeah, you were right. Sorry for removing Jitter in the function. Added back!